esp/scm/ws_resources.ecm

```diff
@@ -25,11 +25,35 @@ ESPStruct HPCCService
     bool TLSSecure;
 };
 
+ESPStruct ServicePorts
+{
+    string Name;
+    string Protocol;
+    unsigned Port;
+    unsigned NodePort;
+    unsigned TargetPort;
+};
+
+//The ExternalIPsEx or ClusterIPsEx is used when ExternalIPs or ClusterIPs is empty.
+//The value is based on the existing https://github.com/kubernetes/kubernetes.
+ESPStruct ServiceConnection
+{
+    string Type;
+    ESParray<string> ExternalIPs;
+    string ExternalIPsEx; //Use this field when ExternalIPs is empty. Set to '<none>' for spec type 'ClusterIP' and spec type 'NodePort';
```

Comment:
Also not sure an Ex ending is right here.  Is this field really helpful if it just tells us that the ips are empty?

Reply:
Maybe call it something like ExternalIPStatus?  You could make it an enum.  We also don't need to follow the kubectl formatting exactly... for example we could remove the '<','>'.

Reply:
Will do